waiter/src/waiter/token.clj

```diff
@@ -321,7 +371,8 @@
                                          make-peer-requests-fn req)
     :get (handle-get-token-request kv-store waiter-hostname req)
     :post (handle-post-token-request clock synchronize-fn kv-store waiter-hostname entitlement-manager
-                                     make-peer-requests-fn validate-service-description-fn req)))
+                                     make-peer-requests-fn validate-service-description-fn req)
+    (utils/map->json-response {:unsupported-method request-method} :status 405)))
```

Comment:
Please use a `try`, `throw` an exception instead, and use `exception->response` in the `catch` block.  Also, please create an integration test.